fwutil/lib.py

```diff
@@ -695,6 +732,43 @@ def update_firmware(self, chassis_name, module_name, component_name):
             log_helper.log_fw_update_end(component_path, firmware_path, False, e)
             raise
 
+    def auto_update_firmware(self, component, boot):
+        chassis_name = component[-6]
+        module_name = component[-5]
+        component_name = component[-4]
+
+        if self.is_modular_chassis():
```

Comment:
if the module_name is N/A,  for modular chassis, it fails since module_component_map is being used

Reply:
good catch! fixed in the latest.